stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/MediaAttachmentContent.kt

```diff
@@ -0,0 +1,485 @@
+                            )
+                        }
+                    }
+                } else {
+                    MediaAttachmentContentItem(
+                        attachment = attachment,
+                        modifier = Modifier.weight(1f),
+                        message = message,
+                        attachmentPosition = attachmentIndex,
+                        onMediaGalleryPreviewResult = onMediaGalleryPreviewResult,
+                        onLongItemClick = onLongItemClick,
+                        playButton = playButton
+                    )
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Displays previews of image and video attachments.
+ *
+ * @param message The original message containing the attachments.
+ * @param attachmentPosition The position of the attachment in the list
+ * of attachments. Used to remember the item position when viewing it in a separate
+ * activity.
+ * @param attachment The attachment that is previewed.
+ * @param onMediaGalleryPreviewResult The result of the activity used for propagating
+ * actions such as media attachment selection, deletion, etc.
+ * @param onLongItemClick Lambda that gets called when the item is long clicked.
+ * @param modifier Modifier used for styling.
+ * @param playButton Represents the play button that is overlaid above video attachment
+ * previews.
+ */
+@Suppress("LongParameterList")
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+internal fun MediaAttachmentContentItem(
+    message: Message,
+    attachmentPosition: Int,
+    attachment: Attachment,
+    onMediaGalleryPreviewResult: (MediaGalleryPreviewResult?) -> Unit,
+    onLongItemClick: (Message) -> Unit,
+    modifier: Modifier = Modifier,
+    playButton: @Composable () -> Unit,
+) {
+    val connectionState by ChatClient.instance().clientState.connectionState.collectAsState()
+    val isImage = attachment.type == AttachmentType.IMAGE
+    val isVideo = attachment.type == AttachmentType.VIDEO
+
+    // Used as a workaround for Coil's lack of a retry policy.
+    // See: https://github.com/coil-kt/coil/issues/884#issuecomment-975932886
+    var retryHash by remember {
+        mutableStateOf(0)
+    }
+
+    val data =
+        if (isImage || (isVideo && ChatTheme.videoThumbnailsEnabled)) {
+            attachment.imagePreviewUrl
+        } else {
+            null
+        }
+
+    val painter = rememberStreamImagePainter(
+        model = ImageRequest.Builder(LocalContext.current)
+            .data(data)
+            .setParameter(key = "retry_hash", value = retryHash)
+            .build()
+    )
+
+    val mixedMediaPreviewLauncher = rememberLauncherForActivityResult(
+        contract = MediaGalleryPreviewContract(),
+        onResult = { result -> onMediaGalleryPreviewResult(result) }
+    )
+
+    // Used to refresh the request for the current page
+    // if it has previously failed.
+    if (data != null && connectionState == ConnectionState.CONNECTED &&
```

Comment:
If you ever use this yourself, keep in mind that it's important to do a nullcheck.

Coil will return `AsyncImagePainter.State.Error` for any request that has passed a `null` as data which means that it does so online as well and this ends up in a recomposition loop. The loop itself is silent an invisible so it's difficult to detect unless you use a recomposition tracker or set a relevant breakpoint.

Reply:
Might be worth noting this somewhere (e.g. in our internal docs)